CDAP-488: added integration tests for ruby

yg2017 opened the pull request:
Branch is ready for review.

artemmakogon committed:
added integration tests for ruby

anwar6953 commented on cdap-stream-clients/ruby/spec/lib/stream-client-ruby/integration_tests_spec.rb:
```diff
@@ -15,16 +14,54 @@ def load_config file
+    writer_test_value = 'test_body'
+    writer = stream_client.create_writer stream_id
+    writer.write writer_test_value
+    data = rest.request('get', "#{stream_id}/events").parsed_response
```
Similar to the integration tests in java or python, we need to specify a time frame, to avoid picking up events that were sent in previous runs of the integration test. Otherwise, even if one execution of the test fails to write to stream, it may still pass because without specifying start/end times, this gets all the events in the stream's history.

Edit: an alternative would be to truncate the stream before each test run, but specifying times for start/end should be good, and will be consistent with other tests (java, python, etc).

anwar6953 commented:
This has a dependency on cdap-authentication-client.
I tried to install that:

```
Alis-MacBook-Pro:ruby alianwar$ pwd
/Users/alianwar/dev/cdap-clients/cdap-authentication-clients/ruby
Alis-MacBook-Pro:ruby alianwar$ sudo rake install
cdap-authentication-client 1.1.0 built to pkg/cdap-authentication-client-1.1.0.gem.
cdap-authentication-client (1.1.0) installed.
```

However, the folder it deploys is empty, and so I was unable to run the test.

```
Alis-MacBook-Pro:ruby alianwar$ cd /Library/Ruby/Gems/2.0.0/gems/
Alis-MacBook-Pro:gems alianwar$ cd cdap-authentication-client-1.1.0/
Alis-MacBook-Pro:cdap-authentication-client-1.1.0 alianwar$ ls
Alis-MacBook-Pro:cdap-authentication-client-1.1.0 alianwar$ 
```

How were you able to resolve this dependency? (locally installing the package did not work for me).

yg2017 commented:
The problem is in the cdap-authentication-client.gemspec, I will fix it.

artemmakogon committed:
Merge remote-tracking branch 'upstream/develop' into feature/CDAP-491-it-ruby

yg2017 commented:
The problem with dependency on cdap-authentication-client was fixed in PR https://github.com/caskdata/cdap-clients/pull/63

yg2017 committed:
improved stream writer test

yg2017 commented:
I've changed stream writer test. Please review.

anwar6953 commented:
https://github.com/caskdata/cdap-clients/pull/63
That PR did fix the problem.

However, I am still having some unexpected test failures problems:
https://s3.amazonaws.com/uploads.hipchat.com/26476/1011205/YM8oHOir3qVQeTJ/Screen%20Shot%202014-11-04%20at%203.34.10%20PM.png

no. 4 is a unit test right, and yet it fails?
(also, no.1 and no.2 might fail because I don't have CDAP runnning with auth/ssl), but no.3 shouldn't be failing at least, right?)

yg2017 committed:
specified promise.rb version

yg2017 commented:
You have weird promise.rb version.
This http://www.rubydoc.info/gems/promise.rb/0.3.0/Promise doesn't have any parameters in initialize.
I have specified the last version 0.6.1 of promise.rb in Gemfile.
Please, check it by executing "bundle list".

yg2017 commented:
I suppose, that problem appears on Ruby 2.0.0. 
I tested on 1.9.3.

wolf31o2 commented:
The default Ruby with Mac OS X starting with Mavericks is 2.0, so it should definitely work, there. I would say a `minimum` of 1.9.3, but definitely needs to work in Ruby 2.0+

yg2017 committed:
added test configuration

yg2017 commented:
The problem was caused by missing "promise.rb" gem. Fixed by running
"bundle list" and "bundle install".

2014-11-06 0:03 GMT+02:00 Chris Gianelloni notifications@github.com:

> The default Ruby with Mac OS X starting with Mavericks is 2.0, so it
> should definitely work, there. I would say a `minimum
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/caskdata/cdap-ingest/pull/95#issuecomment-61889572.

yg2017 commented:
I have added configuration (and updated documentation) for launch tests against different CDAP security modes. Please review.
But first this PR https://github.com/caskdata/cdap-clients/pull/64 should be merged into cdap-clients.

anwar6953 commented:
Ok, yeah the (compatiblility) issue was that the gemfile did not previously specify a version of promise, and so my machine pulled in an different version, and so it did not work.

anwar6953 commented:
The added test configuration looks good, along with most of the PR.

One last concern though, I received a Deprecated Warning when I ran the integration test:
https://s3.amazonaws.com/uploads.hipchat.com/26476/1011205/IuliO45FSInIiqP/Screen%20Shot%202014-11-11%20at%206.28.53%20PM.png

If that is easy to resolve (use the more recent functionality), then can we do that and then merge.
Thanks.

yg2017 committed:
changed deprecated syntax

yg2017 commented:
@anwar6953 I have fixed Deprecated Warning. Please take a look.

anwar6953 commented:
:+1:
